src/convertToObject.js

```diff
@@ -6,7 +6,44 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString === '') {
+    return {};
+  }
+
+  const modifiedStr = sourceString
+    .trim()
+    .split(';')
+    .map((item) => {
+      return item.trim();
+    });
+
+  if (modifiedStr.length === 1) {
+    const [key, value] = sourceString.trim().split(':');
+
+    return {
+      [key]: value.trim(),
+    };
+  }
+
+  let result = {};
+
+  modifiedStr.map((item, i) => {
+    const arr = item.trim().split(':');
+
+    if (i !== modifiedStr.length - 1 && arr[0].length !== 0) {
```

Comment:
The condition `i !== modifiedStr.length - 1` is unnecessary and could lead to incorrect behavior if the last item is a valid key-value pair. Consider removing this condition to ensure all items are processed.